client/dashboard/me/profile-personal-details/update-email/email-verification-banner.tsx

```diff
@@ -0,0 +1,169 @@
+import { resendEmailVerificationMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import { Link } from '@tanstack/react-router';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect } from 'react';
+import Notice from '../../../components/notice';
+import type { UserSettings } from '@automattic/api-core';
+
+// Get email verification params from URL
+function getEmailVerificationParams() {
+	if ( typeof window === 'undefined' ) {
+		return { newEmailResult: null, verified: null };
+	}
+	const params = new URLSearchParams( window.location.search );
+	return {
+		newEmailResult: params.get( 'new_email_result' ),
+		verified: params.get( 'verified' ),
```

Comment:
I was thinking that through the code we use lots of ` === '1'` and `=== '0'`.
Not a requirement for the PR but a thought: Maybe this function could return 4 outputs

```suggestion
		isEmailChangeComplete: params.get( 'new_email_result' ) === '1',
		isEmailVerificationComplete: params.get( 'verified' ) === '1',
        emailChangeFailed: params.get( 'new_email_result' ) === '0',
        emailVerificationFailed: params.get( 'verified' ) === '0',
```

And we use them right where we need them. This way the logic is gated in one single entry point and we don't need to think about what the `1|0` value means.

Reply:
Implemented in [a3f1750](https://github.com/Automattic/wp-calypso/pull/106103/commits/a3f1750231d910b78a2261c28a11ad386304f81a)!